csrc/host_ir/jit.cpp

```diff
@@ -307,7 +497,11 @@ void compile(HostIrJitImpl* pimpl) {
   for (auto* expr : pimpl->container->topLevelExprs()) {
     // TODO: support more expression types
     if (expr->isA<LoadStoreOp>()) {
-      compileLoadStoreOp(expr->as<LoadStoreOp>(), builder, val_to_value);
+      dispatchLoadStoreOp(expr->as<LoadStoreOp>(), builder, val_to_value);
```

Comment:
The fact that you are passing `val_to_value` to every `dispatch*` function is usually a sign that it should be a member variable of a class. For example,

```
class HostIrCompileDispatcher : public OptInDispatch {
 public:
  HostIrCompiler(IrBuilder& builder, std::unordered_map<Val*, llvm::Value*>& val_to_value): builder_(builder), val_to_value_(val_to_value) {}
  void handle(LoadStoreOp*) override;
  void handle(Allocate*) override;
  void handle(Deallocate) override;
  // Not handled instructions automatically trigger an error.
};
```